core/quantity/src/test/Ratio.test.ts

```diff
@@ -0,0 +1,456 @@
+        ratioFormatterSpec.format.precision = entry.precision;
+      const resultRatio = Formatter.formatQuantity(entry.input, ratioFormatterSpec);
+      expect(resultRatio).to.equal(entry.ratio);
+
+      const parserRatioResult = Parser.parseQuantityString(entry.ratio, ratioParser);
+      if (!Parser.isParsedQuantity(parserRatioResult)) {
+        assert.fail(`Expected a parsed from ratio string ${entry.ratio}`);
+      }
+
+      if (null != entry.precision)
+        expect(parserRatioResult.value, `Parsed result for ${entry.ratio} from formatted ${entry.input}`).closeTo(entry.input, 4.999 * (0.1 ** entry.precision));
+      else
+        expect(parserRatioResult.value, `Parsed result for ${entry.ratio} from formatted ${entry.input}`).closeTo(entry.input, 4.999 * (0.1 ** defaultPrecision));
+
+    }
+  }
+
+  describe("RatioType Tests", () => {
+    it("OneToN", async () => {
+      const testData: { input: number, ratio: string }[] = [
+        { input: 1.0, ratio: "1:1" },
+        { input: 2.0, ratio: "1:0.5" },
+        { input: 0.5, ratio: "1:2" },
+        { input: 0.333, ratio: "1:3.003" },
+        { input: 0.3333, ratio: "1:3" },
+        { input: 0.2857, ratio: "1:3.5" },
+        { input: 0.25, ratio: "1:4" },
+        { input: 0.6667, ratio: "1:1.5" },
+      ];
+      await testRatioType("OneToN", testData);
+    });
+
+    it("NToOne", async () => {
+      const testData: { input: number, ratio: string }[] = [
+        { input: 1.0, ratio: "1:1" },
+        { input: 2.0, ratio: "2:1" },
+        { input: 0.5, ratio: "0.5:1" },
+        { input: 0.333, ratio: "0.333:1" },
+        { input: 0.3333, ratio: "0.333:1" },
+        { input: 0.2857, ratio: "0.286:1" },
+        { input: 0.25, ratio: "0.25:1" },
+        { input: 0.6667, ratio: "0.667:1" },
+      ];
+      await testRatioType("NToOne", testData);
+    });
+
+    it("ValueBased", async () => {
+      const testData: { input: number, ratio: string }[] = [
+        { input: 1.0, ratio: "1:1" },
+        { input: 2.0, ratio: "2:1" },
+        { input: 0.5, ratio: "1:2" },
+        { input: 0.333, ratio: "1:3.003" },
+        { input: 0.3333, ratio: "1:3" },
+        { input: 0.2857, ratio: "1:3.5" },
+        { input: 3.5, ratio: "3.5:1" },
+        { input: 0.25, ratio: "1:4" },
+        { input: 4, ratio: "4:1" },
+        { input: 0.6667, ratio: "1:1.5" },
+
+      ];
+      await testRatioType("ValueBased", testData);
+    });
+
+    it("UseGreatestCommonDivisor", async () => {
+      const testData: { input: number, ratio: string }[] = [
+        { input: 1.0, ratio: "1:1" },
+        { input: 2.0, ratio: "2:1" },
+        { input: 0.5, ratio: "1:2" },
+        { input: 0.333, ratio: "333:1000" },
+        { input: 0.3333, ratio: "333:1000" },
+        { input: 0.2857, ratio: "143:500" },
+        { input: 0.25, ratio: "1:4" },
+        { input: 0.6667, ratio: "667:1000" },
+      ];
+      await testRatioType("UseGreatestCommonDivisor", testData);
+    });
```

Comment:
it seems like there would be some interesting cases here with varying precision that aren't covered by the tests below